public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -276,39 +299,59 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
         }
 
 
-
         List<String> domainIds = null;
         if (searchConceptsRequest.getDomain() != null) {
             domainIds = DOMAIN_MAP.get(searchConceptsRequest.getDomain()).asList();
         }
 
         ConceptService.StandardConceptFilter convertedConceptFilter = ConceptService.StandardConceptFilter.valueOf(standardConceptFilter.name());
 
+
+        List<Concept> conceptSynonymList = null;
+        List<Long> synonymConceptIds = new ArrayList<>();
+        if(searchConceptsRequest.getQuery() != null && !searchConceptsRequest.getQuery().isEmpty()){
+            conceptSynonymList = conceptDao.findConceptSynonyms(ConceptService.modifyMultipleMatchKeyword(searchConceptsRequest.getQuery()));
+            for(Concept c:conceptSynonymList){
+                synonymConceptIds.add(c.getConceptId());
+            }
+        }
+
         Slice<Concept> concepts =
-                conceptService.searchConcepts(searchConceptsRequest.getQuery(), convertedConceptFilter,
-                        searchConceptsRequest.getVocabularyIds(), domainIds, maxResults, minCount);
-        ConceptListResponse response = new ConceptListResponse();
-        List<Concept> matchedConcepts = concepts.getContent();
+                    conceptService.searchConcepts(searchConceptsRequest.getQuery(), convertedConceptFilter,
+                            searchConceptsRequest.getVocabularyIds(), domainIds, maxResults, minCount, synonymConceptIds);
+            ConceptListResponse response = new ConceptListResponse();
+            List<Concept> matchedConcepts = concepts.getContent();
+            List<String> conceptSynonymNames = new ArrayList<>();
+
+            for(Concept con : matchedConcepts){
+                String conceptCode = con.getConceptCode();
+                String conceptId = String.valueOf(con.getConceptId());
+
+                if(con.getSynonyms() != null){
+                    response.setMatchType(MatchType.SYNONYM);
```

Comment:
Since each concept_name is also a synonm -- we need more here to have this be accurate... basically you would have to see if the query string is actually in the concept_name ... If it is , than it is a name match , if it isn't then it is a synonm match.  I think I would just call it a name match for now.  People may not be ready to handle another match type on the front end.

Reply:
Addressed.